p4_symbolic/symbolic/symbolic.h

```diff
@@ -44,6 +45,12 @@ namespace symbolic {
 // evaluation.
 z3::context &Z3Context();
 
+// This struct contains any metadata or information that needs to be passed
```

Comment:
Why create a struct with just one entry? I think I'd prefer not to have the indirection, so when I read the code, I don't have to jump to this definition to know what's actually going on. If this is in anticipation of having more fields in the future, let's do it in the future when the usecase arrises.

Reply:
ok makes sense.